src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            var root = (await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false))!;
+
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localFunction, cancellationToken);
+            var parameters = CreateParameterSymbols(captures);
+
+            // Finds all the call sites of the local function
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(
+                localFunctionSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            // Now we need to find all the refereces to the local function that we might need to fix.
+            var shouldWarn = false;
+            var invocationReferences = new List<InvocationExpressionSyntax>();
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                foreach (var location in referencedSymbol.Locations)
+                {
+                    // Since this is a local function, all reference must be in the same tree.
+                    var referenceNode = root.FindNode(location.Location.SourceSpan);
+                    if (!(referenceNode is IdentifierNameSyntax identifierNode))
+                    {
+                        continue;
+                    }
+
+                    if (identifierNode.Parent is InvocationExpressionSyntax invocation)
+                    {
+                        invocationReferences.Add(invocation);
+                    }
+                    else
+                    {
+                        // We won't be able to fix non-invocation references, 
+                        // e.g. creating a delegate. 
+                        shouldWarn = true;
+                    }
+                }
+            }
+
+            var nodeToTrack = new List<SyntaxNode>(invocationReferences) { localFunction };
+            root = root.TrackNodes(nodeToTrack);
+
+            // Fix all invocations by passing in additional arguments.
+            foreach (var invocation in invocationReferences.OrderByDescending(n => n.Span.Start))
+            {
+                var currentInvocation = root.GetCurrentNode(invocation);
+
+                var seenNamedArgument = currentInvocation.ArgumentList.Arguments.Any(a => a.NameColon != null);
+                var seenDefaultArgumentValue = currentInvocation.ArgumentList.Arguments.Count < localFunction.ParameterList.Parameters.Count;
+
+                var newArguments = parameters.Select(
+                    p => CSharpSyntaxGenerator.Instance.Argument(
+                        seenNamedArgument || seenDefaultArgumentValue ? p.Name : null,
+                        p.RefKind,
+                        p.Name.ToIdentifierName()) as ArgumentSyntax);
+
+                var newArgList = currentInvocation.ArgumentList.WithArguments(currentInvocation.ArgumentList.Arguments.AddRange(newArguments));
+                var newInvocation = currentInvocation.WithArgumentList(newArgList);
```

Comment:
>var newInvocation = currentInvocation.WithArgumentList(newArgList); [](start = 16, length = 67)

Will this correctly fix nested invocations: `A(..., A(...), ...);`?

Reply:
I think this test covers this scenario

https://github.com/dotnet/roslyn/pull/37629/files/2aae01251486454f9da27039d5874079d5c15266#diff-af850fe0d00ca594c43bd4079d7b2553R245

Reply:
Ah, thanks!

---
In reply to: [324867430](https://github.com/dotnet/roslyn/pull/37629#discussion_r324867430) [](ancestors = 324867430)